libkineto/src/CuptiActivityProfiler.cpp

```diff
@@ -800,6 +800,18 @@ void CuptiActivityProfiler::finalizeTrace(const Config& config, ActivityLogger&
     logger.handleResourceInfo(resource, captureWindowStartTime_);
   }
 
+  for (auto &session : sessions_){
+    auto device_info = session->getDeviceInfo();
+    if (device_info != nullptr){
+      logger.handleDeviceInfo(*device_info, captureWindowStartTime_);
+    }
+
+    auto resource_infos = session->getResourceInfos();
```

Comment:
similar to above

Reply:
thanks for the comment, but the situation here is different: getResourceInfos returns vector